CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,187 @@
+1. Proposed - The exchange has been proposed, but not yet approved by the approver. The proposer may still cancel their proposal and be refunded their deposit.
+2. Approved - The exchange has been proposed and approved by the approver, but not yet executed or vetoed. The proposer may no longer cancel their proposal at this point.
+3. Executed - The exchange has been proposed, approved by the approver, and executed.
+4. Cancelled - The exchange has been cancelled by the proposer when it was in the Proposed state, or it has been vetoed by Governance when it was in the Approved stage.
+
+### Specific Changes:
+
+#### Modifications of existing contracts:
+
+##### `StableToken.sol`
+
+Currently, `StableToken.sol` only allows its `Exchange.sol` or `Validators.sol` to [mint](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L225) new stable tokens. Similarly, only its `Exchange.sol` can [burn](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L275) stable tokens.
+
+The following modifications will be made:
+1. **Modification to [mint()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L224)**
+   * Allow `msg.sender` to be `GrandaMento` in addition to the existing permitted senders, Exchange and `Validators.sol`.
+2. **Modification to [burn()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L273)**
+   * Allow `msg.sender` to be `GrandaMento` in addition to the existing permitted sender, the Exchange.
+
+#### New contracts:
+
+##### `GrandaMento.sol`
+
+A new contract, `GrandaMento.sol`, is created, added to the Registry with the identifier `GrandaMento`, and owned by Governance. `GrandaMento` is a singleton contract that is capable of facilitating exchanges for multiple stable tokens-- that is, with the approval of Governance, `GrandaMento` could support `CELO <-> cUSD`, `CELO <-> cEUR`, and future `CELO <-> cXXX` exchanges. `GrandaMento` must be a spender of `Reserve.sol`, and supported stable tokens must allow `GrandaMento` to mint/burn them.
+
+The contract has the following configurable parameters:
+
+1. **`address public approver;`** - Set by Governance. Intended to be a multisig and has the authority to approve exchanges. The signers for the multisig are at the discretion of Governance, but it would comprise of community members or members on behalf of organizations that are aligned with the Celo network/ecosystem.
+2. **`uint256 public exchangeWaitPeriodSeconds;`** - Set by Governance. The minimum amount of time in seconds that must elapse between a proposed exchange being approved and when the exchange can be executed. Should give sufficient time for Governance to veto an approved exchange.
+3. **`FixidityLib.Fraction public spread;`** - Set by Governance. The percent fee imposed upon an exchange execution.
```

Comment:
Do we have any consensus about how much that fee should be?

Reply:
I think we didn't talk much about this yet (maybe @tkporter and @albertclabs have more details meanwhile). @albertclabs and I had a brief chat about it and he mentioned users would probably not care about fees like 2-3 % what they would get OTC, which would be a neat way to bolster the reserve. My thoughts were at least the costs (fees + slippage) that they would have with Mento when they would execute slowly in batches. What do you think?

Reply:
My feeling here is that 0.5% similar to Mento today would be appropriate. It could likely be higher up to 1% and still be similar to what they would experience writing a block purchasing script. I'd shy away from much more than that as the primary aim of this is not the bolster the Reserve, but to increase liquidity in the greater ecosystem.

Reply:
Seems like 0.75% would be a good number to present to the community?

Reply:
It looks like most of the discussion has been around the amount of the fee (as something desired to be greater than zero), and not the inclusion of this parameter. We can discuss going into the CGP what value to set it to, as it looks like there's broad agreement this should be a governable value.